pkg/project/kubernetes.go

```diff
@@ -0,0 +1,142 @@
+// or by creating directly a `namespace` resource.
+// With the `wait` flag, the function will wait for the `default` service account
+// to be created in the namespace before to return.
+func (o kubernetesClient) Create(projectName string, wait bool) error {
+	if projectName == "" {
+		return errors.Errorf("no project name given")
+	}
+
+	projectSupport, err := o.client.IsProjectSupported()
+	if err != nil {
+		return errors.Wrap(err, "unable to detect project support")
+	}
+	if projectSupport {
+		err = o.client.CreateNewProject(projectName, wait)
+		if err != nil {
+			return errors.Wrap(err, "unable to create new project")
+		}
+
+	} else {
+		_, err = o.client.CreateNamespace(projectName)
+		if err != nil {
+			return errors.Wrap(err, "unable to create new project")
+		}
+	}
+
+	if wait {
+		err = o.client.WaitForServiceAccountInNamespace(projectName, "default")
+		if err != nil {
+			return errors.Wrap(err, "unable to wait for service account")
+		}
+	}
+	return nil
+}
+
+// Delete deletes the project (the `project` resource if supported, or directly the `namespace`)
+// with the name projectName and returns an error if any
+func (o kubernetesClient) Delete(projectName string, wait bool) error {
+	if projectName == "" {
+		return errors.Errorf("no project name given")
+	}
+
+	projectSupport, err := o.client.IsProjectSupported()
+	if err != nil {
+		return errors.Wrap(err, "unable to detect project support")
+	}
+
+	if projectSupport {
+		// Delete the requested project
+		err := o.client.DeleteProject(projectName, wait)
+		if err != nil {
+			return errors.Wrapf(err, "unable to delete project %s", projectName)
+		}
+	} else {
+		err := o.client.DeleteNamespace(projectName, wait)
+		if err != nil {
+			return errors.Wrapf(err, "unable to delete namespace %s", projectName)
+		}
+	}
+	return nil
+}
+
+// List all the projects on the cluster and returns an error if any
+func (o kubernetesClient) List() (ProjectList, error) {
+	currentProject := o.client.GetCurrentNamespace()
+
+	projectSupport, err := o.client.IsProjectSupported()
+	if err != nil {
+		return ProjectList{}, errors.Wrap(err, "unable to detect project support")
+	}
+
+	var allProjects []string
+	if projectSupport {
+		allProjects, err = o.client.ListProjectNames()
+		if err != nil {
+			return ProjectList{}, errors.Wrap(err, "cannot get all the projects")
+		}
+	} else {
+		allProjects, err = o.client.GetNamespaces()
+		if err != nil {
+			return ProjectList{}, errors.Wrap(err, "cannot get all the namespaces")
+		}
+	}
+
+	projects := make([]Project, len(allProjects))
+	for i, project := range allProjects {
+		isActive := project == currentProject
+		projects[i] = NewProject(project, isActive)
+	}
+
+	return NewProjectList(projects), nil
+}
+
+// Exists checks whether a project with the name `projectName` exists and returns an error if any
+func (o kubernetesClient) Exists(projectName string) (bool, error) {
+	projectSupport, err := o.client.IsProjectSupported()
```

Comment:
Why not have the check for empty value of `projectName` here? We have it in `Create` and `Delete` methods. 🤔

Reply:
I think that this specific case is detected by `GetProject`/`GetNamespace`